File: backend/app/schemas/maintenance.py
from pydantic import BaseModel, Field
from datetime import datetime
from enum import Enum


class MaintenanceStatus(str, Enum):
    pending = "pending"
    approved = "approved"
    rejected = "rejected"
    completed = "completed"


class MaintenanceCreate(BaseModel):
    vehicle_id: int
    maintenance_type: str
    description: str = Field(max_length=400)
    date: datetime
    estimated_cost: float
    workshop: str | None = None


class MaintenanceUpdateStatus(BaseModel):
    status: MaintenanceStatus


class MaintenanceOut(BaseModel):
    id: int
    dispatcher_id: int
    vehicle_id: int
    maintenance_type: str
    description: str
    date: datetime
    estimated_cost: float
    workshop: str | None
    status: MaintenanceStatus

    class Config:
        from_attributes = True

File: frontend/src/api/maintenance.js
import { apiFetch } from './client'

/**
 * Maintenance shape from backend (MaintenanceOut):
 * { id, dispatcher_id, vehicle_id, maintenance_type, description,
 *   date, estimated_cost, workshop, status }
 *
 * Status values: pending | approved | rejected | completed
 */

/** GET /maintenance/  →  MaintenanceOut[]  (manager only) */
export const listMaintenanceApi = () => apiFetch('/maintenance/')

/** GET /maintenance/me  →  MaintenanceOut[]  (dispatcher — own requests) */
export const myMaintenanceApi = () => apiFetch('/maintenance/me')

/** POST /maintenance/  →  MaintenanceOut  (dispatcher only) */
export const createMaintenanceApi = (data) =>
  apiFetch('/maintenance/', {
    method: 'POST',
    body: JSON.stringify(data),
  })

/** PATCH /maintenance/:id/status  →  MaintenanceOut  (manager only) */
export const updateMaintenanceStatusApi = (id, status) =>
  apiFetch(`/maintenance/${id}/status`, {
    method: 'PATCH',
    body: JSON.stringify({ status }),
  })
